library/azure_rm_resource.py

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Zim Kalinowski, <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_resource
+version_added: "2.6"
+short_description: Create any Azure resource.
+description:
+  - Create, update or delete any Azure resource using Azure REST API.
+  - Refer to https://docs.microsoft.com/en-us/rest/api/ regarding details related to specific resource REST API.
+
+options:
+  url:
+    description:
+      - Azure RM Resource URL.
+  provider:
+    description:
+      - Provider type, should be specified in no URL is given
+  resource_group:
+    description:
+      - Resource group to be used, should be specified if needed and URL is not specified
+  resource_type:
+    description:
+      - Resource type, should be valid for specified provider
+  resource_name:
+    description:
+      - Resource name, should be specified if needed and URL is not specified
+  subresource:
+    description:
+      - List of subresources
+    type: list
+    suboptions:
+      namespace:
+        description:
+          - Subresource namespace
+      type:
+        description:
+          - Subresource type
+      name:
+        description:
+          - Subresource name
+  body:
+    description:
+      - The body of the http request/response to the web service.
+  method:
+    description:
+      - The HTTP method of the request or response. It MUST be uppercase.
+        choices: [ "GET", "PUT", "POST", "HEAD", "PATCH", "DELETE", "MERGE" ]
+        default: "PUT"
+  status_code:
+    description:
+      - A valid, numeric, HTTP status code that signifies success of the
+        request. Can also be comma separated list of status codes.
+    default: [ 200, 201, 202 ]
+  state:
+    description:
+      - Assert the state of the resource. Use C(present) to create or update resource or C(absent) to delete resource.
+    default: present
+    choices:
+        - absent
+        - present
+
+extends_documentation_fragment:
+  - azure
+
+author:
+  - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+  - name: Update scaleset info using azure_rm_resource
```

Comment:
do you want to add some sentences or samples about using this module could provide azure resources which hasn't delegated module yet?

Reply:
maybe a good idea